tests/rptest/tests/partition_balancer_test.py

```diff
@@ -139,7 +140,20 @@ def wait_until_status(self, predicate, timeout_sec=120):
         def check():
             req_start = time.time()
 
-            status = admin.get_partition_balancer_status(timeout=10)
+            def get_partition_balancer_status():
+                try:
+                    status = admin.get_partition_balancer_status(timeout=10)
+                    return True, status
+                except RetryError:
+                    return False
+
+            status = wait_until_result(
```

Comment:
We already have retries inside the admin API wrapper. I don't think we need two-level retries, we can just make some parameters for the [Retry object](https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html?highlight=Retry#urllib3.util.Retry) configurable via the `Admin` constructor.

BTW how can we be sure that the current number of retries is enough? How long was redpanda without the controller leader in the failed test run and for how long did we retry the request?

Reply:
agree about avoiding two levels of retry.

Reply:
Previously we had 5 retries with backoff_factor = 1
It means that we had 5 retries with wait for 0, 1, 2, 3, 4 seconds between attempts.

In new version I have added 15 retries. It mean that we will wait for 0, 1, .. 14 seconds between attempts.
Max total wait is 105 sec.

Reply:
I think it grows exponentially and then is capped at 120s (unfortunately not tunable in urrlib3 v1). So it is more like 0,1,2,4,..120,..

Reply:
Yeah, you are right. Formula from docs: {backoff factor} * (2 ** ({number of total retries} - 1))
Maybe 15 is too much for this